enterprise/server/remote_execution/runner/runner.go

```diff
@@ -129,6 +129,9 @@ type CommandRunner struct {
 	// runner.
 	InstanceName string
 
+	// pool holds a reference to the parent pool that constructed this runner.
+	pool *Pool
```

Comment:
this seems fishy (needing a back reference to the parent)

Can we refactor the pool code to just ask all runners if they are active / inactive as necessary so that state only lives in one place?

Reply:
Done